Code_Review_Assignment_JrSDET.java

```diff
@@ -1 +1,172 @@
+package com.org.happyfox;
+
+import java.awt.AWTException;
+import java.awt.Robot;
+import java.awt.event.KeyEvent;
+import java.util.concurrent.TimeUnit;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.interactions.Actions;
+
+public class Testcase101 {
+
+    public static void main(String[] args) throws InterruptedException, AWTException {
+        System.setProperty("webdriver.gecko.driver", "C:\\Users\\Johny\\Downloads\\geckodriver-v0.33.0-win64\\geckodriver.exe");
+        WebDriver driver = new FirefoxDriver();
+        driver.get("https://interview.supporthive.com/staff/");
+        driver.manage().timeouts().implicitlyWait(30, TimeUnit.SECONDS);
+        driver.manage().window().maximize();
+        driver.findElement(By.id("id_username")).sendKeys("Agent");
+        driver.findElement(By.id("id_password")).sendKeys("Agent@123");
+        driver.findElement(By.id("btn-submit")).click();
+        WebElement tickets = driver.findElement(By.id("ember29"));
+        Actions action = new Actions(driver);
+        action.moveToElement(tickets).build().perform();
+        WebElement statuses = driver.findElement(By.linkText("Statuses"));
+        statuses.click();
+        driver.findElement(By.xpath("/html/body/div[3]/div/section/section/div/header/button")).click();
+        driver.findElement(By.tagName("input")).sendKeys("Issue Created");
+        WebElement statusColourSelect = driver.findElement(By.xpath("//div[@class='sp-replacer sp-light']"));
+        statusColourSelect.click();
+        WebElement statusColourEnter = driver.findElement(By.xpath("//input[@class='sp-input']"));
+        statusColourEnter.clear();
+        statusColourEnter.sendKeys("#47963f");
+        Robot r = new Robot();
+        r.keyPress(KeyEvent.VK_ESCAPE);
+
+        WebElement firstElement = driver.findElement(By.xpath("//a[@id='first-link']"));
+        firstElement.click();
+
+        WebElement secondElement = driver.findElement(By.xpath("//a[@id='second-link']"));
+        secondElement.click();
+
+        driver.findElement(By.tagName("textarea")).sendKeys("Status when a new ticket is created in HappyFox");
+        WebElement addCreate = driver.findElement(By.xpath("//button[@class ='hf-entity-footer_primary hf-primary-action ember-view']"));
+        addCreate.click();
+
+        Thread.sleep(3000);

```

Comment:
Instead of thread.sleep, we can use ImplicitlyWait here.

Disadvantage of thread.sleep() -> This thread is completely blocked and will not handle any other task until the sleep is over